src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1284,7 +1284,7 @@ ss::future<end_tx_reply> tx_gateway_frontend::do_end_txn(
     // is active
     if (stm->gate().is_closed()) {
         return ss::make_ready_future<end_tx_reply>(
-          end_tx_reply{.error_code = tx_errc::unknown_server_error});
+          end_tx_reply{.error_code = tx_errc::not_coordinator});
```

Comment:
Maybe return better `coordinator_not_available`? It means in kafka that `if the coordinator receiving the request is being shutdown.`

Reply:
ok, it's a retryable error so it'll work too, changed